src/orange/construct/OrangeInputIO.json.cc

```diff
@@ -192,8 +186,22 @@ void from_json(nlohmann::json const& j, RectArrayInput& value)
 
         for (auto i : range(daughters.size()))
         {
-            value.daughters[parents[i]] = {
-                UniverseId{daughters[i]}, slice<3>(make_span(translations), i)};
+            DaughterInput daughter;
+            daughter.universe_id = UniverseId{daughters[i]};
+
+            // Read and convert transform
+            Translation tr{slice<3>(make_span(translations), i)};
+            if (CELER_UNLIKELY(tr.translation() == (Real3{0, 0, 0})))
+            {
+                daughter.transform = NoTransformation{};
+            }
+            else
+            {
+                daughter.transform = tr;
+            }
```

Comment:
Could you use `make_transform()` here?
```suggestion
            daughter.transform
                = detail::make_transform(slice<3>(make_span(translations), i));
```

Reply:
Yes exactly, thank you. This file has been rebased too many times 😔